osu.Game/Skinning/BeatmapSkinProvidingContainer.cs

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Game.Configuration;
+
+namespace osu.Game.Skinning
+{
+    /// <summary>
+    /// A container which overrides existing skin options with beatmap-local values.
+    /// </summary>
+    public class BeatmapSkinProvidingContainer : SkinProvidingContainer
+    {
+        private readonly Bindable<bool> beatmapSkins = new Bindable<bool>();
+        private readonly Bindable<bool> beatmapHitsounds = new Bindable<bool>();
+
+        protected override bool AllowConfigurationLookup => beatmapSkins.Value;
+        protected override bool AllowDrawableLookup(string componentName) => beatmapSkins.Value;
```

Comment:
Seems to break beatmap skins on first load (if you uncheck-check the checkbox it'll work). Probably due to load() order of the hierarchy.

Reply:
Fix via loading `DrawableRuleset` in a second pass.